torch_geometric/graphgym/models/layer.py

```diff
@@ -1,16 +1,67 @@
+import copy
+from dataclasses import dataclass
+
 import torch
 import torch.nn as nn
 import torch.nn.functional as F
 import torch_geometric as pyg
 
-from torch_geometric.graphgym.config import cfg
 from torch_geometric.graphgym.models.act import act_dict
 from torch_geometric.graphgym.contrib.layer.generalconv import (
     GeneralConvLayer, GeneralEdgeConvLayer)
 
 import torch_geometric.graphgym.register as register
 
 
+@dataclass
```

Comment:
Is there a difference between `@dataclass` and simply using `NamedTuple`s?

Reply:
Data class is a class to store data with attribute while named tuple is a dictionary under the hood.
https://www.geeksforgeeks.org/difference-between-dataclass-vs-namedtuple-vs-object-in-python/